x-pack/legacy/plugins/siem/public/components/field_renderers/field_renderers.tsx

```diff
@@ -132,89 +132,90 @@ export const hostNameRenderer = (host: HostEcsFields, ipFilter?: string): React.
 export const whoisRenderer = (ip: string) => <WhoIsLink domain={ip}>{i18n.VIEW_WHOIS}</WhoIsLink>;
 
 export const reputationRenderer = (ip: string): React.ReactElement => (
-  <>
-    <VirusTotalLink link={ip}>{i18n.VIEW_VIRUS_TOTAL}</VirusTotalLink>
-    {', '}
-    <ReputationLink domain={ip}>{i18n.VIEW_TALOS_INTELLIGENCE}</ReputationLink>
-  </>
+  <ReputationLink domain={ip} />
 );
 
-interface DefaultFieldRendererProps {
-  rowItems: string[] | null | undefined;
+interface DefaultFieldRendererProps<T = string> {
+  rowItems: T[] | null | undefined;
```

Comment:
To get some help from the compiler to ensure that only input that can really be rendered is passed in, is it possible to narrow the types here, and everywhere else where `string[]` was changed to `T[]`? 

For example, would changing `T` with `React.ReactNode,` or whitelisting specific types, e.g.  `rowItems: string[] | Foo[] | Bar[] | null | undefined;` work here, and throughout the rest of this PR where `string[]` was changed to `T[]`?

Reply:
I was considering using `string | ReputationLinkSetting` instead of T, but found this component is reused by embeddable, so I held back as not sure I should put ReputationLinkSetting this hight up, as it is only the case for `zeekrenderer`.